### PR TITLE
Support for CORS

### PR DESCRIPTION
**Pull request by mkjmdski**
I've noticed that simple_bucket is missing support for CORS so I decided to add it.

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
Running terraform validate
terraform_validate . 
Success!
 The configuration is valid.
terraform_validate ./examples/multiple_buckets 
Success!
 The configuration is valid.
terraform_validate ./examples/simple_bucket 
Error: 
Reference to undeclared input variable
  on ../../modules/simple_bucket/main.tf line 46, in resource "google_storage_bucket" "bucket":
  46:     for_each = 
var.corse
 == null ? [] : [var.corse]
An input variable with the name "corse" has not been declared. Did you mean
"cors"?
Error: 
Reference to undeclared input variable
  on ../../modules/simple_bucket/main.tf line 46, in resource "google_storage_bucket" "bucket":
  46:     for_each = var.corse == null ? [] : [
var.corse
]
An input variable with the name "corse" has not been declared. Did you mean
"cors"?
terraform_validate ./modules/simple_bucket 
Error: 
Reference to undeclared input variable
  on main.tf line 46, in resource "google_storage_bucket" "bucket":
  46:     for_each = 
var.corse
 == null ? [] : [var.corse]
An input variable with the name "corse" has not been declared. Did you mean
"cors"?
Error: 
Reference to undeclared input variable
  on main.tf line 46, in resource "google_storage_bucket" "bucket":
  46:     for_each = var.corse == null ? [] : [
var.corse
]
An input variable with the name "corse" has not been declared. Did you mean
"cors"?
terraform_validate ./test/fixtures/multiple_buckets 
Success!
 The configuration is valid.
terraform_validate ./test/setup 
Success!
 The configuration is valid.
 ```

**Comment by mkjmdski** [comment]
@morgante  could you review this too ?

File: modules/simple_bucket/variables.tf [hunk 1]
```diff
@@ -78,6 +78,17 @@ variable "retention_policy" {
   default = null
 }
 
+variable "cors" {
+  description = "Configuration of CORS for bucket with structure as defined in https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_bucket#cors."
+  type = list(object({
```
Comment (bharathkkb) [inline modules/simple_bucket/variables.tf:83]: we should use `any` and look up. This way users won't have to provide empty values for unused fields. [Example](https://github.com/terraform-google-modules/terraform-google-cloud-storage/blob/a0982c94f92df37d903af0a85e7ee5fe1e655d1e/variables.tf#L192)

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR @mkjmdski

**Comment by mkjmdski** [comment]
hi @bharathkkb  I think this PR is ready to review again :)

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**